### ISSUE TITLE
How to derive the actual number of words per line for each chapter?

### ISSUE THREAD

**Issue by AlGantori** [quran/quran.com-images#36]
If I understand the main page description these "scripts" render from a font an image of the page then builds the rectangle bounds for each words (glyphs) generated (Correct?)

Does it also build one line bitmap at a time for a 15 lines/rows per page madina mushaf?

It sounds overly complex if all I want is the word count per line for each chapter.
Something like the following (showing word count for Fatiha, Baqara,)
```
   "1": [4,5,4,4,4,5,3],
   "2": [7,5,4,8,6,6,
        9,9,7,9,8,8,9, 8,9,10,9,10,8,7, 7, ..... ],
```

**Comment by ahmedre** [comment]
yes, your understanding is correct. and yes, it builds one line at a time.
for what you want to do, i'd download and get the database from this repo and then get this data with a query (or with a script that just does this for each page). if i recall, the table here should contain the line information as well.

**Comment by AlGantori** [comment]
Are you sure this kind of data is not already available in some XML/JSON resource?

I have done indirectly some node.js based development but I don't recognize the commands installation notes  like the following:
```
ppm install dmake
ppm install dbd-mysql
ppm install yaml
```
are these expected to be executed inside some CLI? or some linux distro?
Thanks for helping out because at this point I am clueless.
I am running in Windows7

**Comment by ahmedre** [comment]
you don't need to do any of those commands nor run this script itself - just download the database and import it and write a script yourself.

**Comment by AlGantori** [comment]
By database you mean download the sql folder in this repo.

I have MySQL Workbench, it's a beast I never got acquainted with all of its terms Open Model, ???
It seems oriented to open dbs over some network connection, I am having hard time making it open a local file.  It managed to open schema.mwb and throws me into the err diagram mode, I want to see the tables and data.

Which of these files should I be attempting to open?
![image](https://user-images.githubusercontent.com/11699920/80863153-2c220500-8c2f-

11ea-8982-846df1dba889.png)

Would you suggest a better tool than MySQL Workbench 5.2.44 CE?

By me writing scrips you mean write SQL queries to retrieve info, perhaps from glyph_line_page table?

Thank you for holding my hands thru this.

**Comment by AlGantori** [comment]
- So after upgrading MySQL Workbench to 8.0.20 it quit working and has error, these companies Oracle, Microsoft just ruin any piece of software they grab and make it a monster.  At the end I just ditched this garbage software installing so much junk on my system.
- I have not worked with SQL from like 1987 Sybase SQL prior to MS SQL
- I guess you have to have a local server installed.
- So I installed HeidiSQL and XAMPP as the server.
- Was able to open the 02-database.sql

- Is the **glyph_ayah** the table I should be deriving word count from?
Since I am purely guessing can you confirm the row below refers to the first verse of Fatiha in this case Basmallh and the row is actually the verse number which in my case I won't be counting as a word.

![image](https://user-images.githubusercontent.com/11699920/80893478-80f16a00-8c87-11ea-93d0-960b0b9ad467.png)

Will tajweed markings (eg. small-meen etc..) be appearing as separate rows in this table or lumped with the previous word (as a single glyph)?

I feel like this is terrible, I would have to query and group count on ayah_number and minus one for the aya_number (hindi thingy) to get my word count???

**Comment by AlGantori** [comment]
Oh that table does not give the line_number
so for this line wrapping
![image](https://user-images.githubusercontent.com/11699920/80894155-8fdb1b00-8c8d-11ea-869d-a837660e5a71.png)

This would be its data, basically 9 tokens not including the verse_number but including the 2 tajweed markers for this particular 1st line of chapter2

![image](https://user-images.githubusercontent.com/11699920/80894193-ddf01e80-8c8d-11ea-838a-0c49507de1c5.png)

**Comment by AlGantori** [comment]
Specifically for Page#2 this database is about this particular layout

![image](https://user-images.githubusercontent.com/11699920/80895029-5b6b5d00-8c95-11ea-849a-cfac8f1e415b.png)


Matching query
```
SELECT COUNT(line_number) FROM `glyph_page_line` WHERE page_number = 2 GROUP BY line_number;
```
![image](https://user-images.githubusercontent.com/11699920/80895038-72aa4a80-8c95-11ea-8dbf-2a0eb7019635.png)

I happen to be working with the Tajweed version page2 is a bit different, that's ALRIGHT I will handle that.

Again my current road block is detecting a token is a verse number???

**Comment by ahmedre** [comment]
the glyph table will tell you what "type" the glyph is - so you can exclude the ayah markers that way.

**Comment by AlGantori** [comment]
Wow I can't believe I am doing a 3 way join to get this, it appear all verse-numbers are typed as "end"

![image](https://user-images.githubusercontent.com/11699920/80896002-6bd40580-8c9e-11ea-87df-ab63d04e021d.png)

![image](https://user-images.githubusercontent.com/11699920/80896013-80180280-8c9e-11ea-812a-ccfbca5f613a.png)

Mission almost accomplished  !!! ALLAH AKBAR !!!

**Comment by ahmedre** [comment]
awesome al7amdulillah! make sure to not include other things like pauses (so just include words).

**Comment by AlGantori** [comment]
I my current hacking I am including the Tajweed-marker (here a pause) and verse-numbers it's just that my algorithm is based on word count.   I probably should use a similar approach in my JSON expand it perhaps and tag the tokens with their type per even sub-type.  I would not be calling verse-number "end" as I want to flow them either as aaya-prefix (default) or suffix in future renderings.
